### PR TITLE
TOOL-20 Updated dependencies to use new bundle id resolver.

### PR DESCRIPTION
**Pull request by richard-bogdan-bitrise**

### CODE REVIEW

File: utils/entitlements.go [hunk 1]
```diff
@@ -75,7 +74,7 @@ func ProjectEntitlementsByBundleID(pth, schemeName, configurationName string) (m
 		return nil, fmt.Errorf("target not found: %s", archiveEntry.BuildableReference.BlueprintIdentifier)
 	}
 
-	targets := append([]xcodeproj.Target{mainTarget}, mainTarget.DependentExecutableProductTargets()...)
+	targets := append([]xcodeproj.Target{mainTarget}, mainTarget.DependentExecutableProductTargets(false)...)
```
Comment (lszucs) [inline utils/entitlements.go]: Can you elaborate on why was the `false` parameter introduced?

Comment (richard-bogdan-bitrise) [reply utils/entitlements.go]: This was the given method before the dep update:
https://github.com/bitrise-steplib/steps-xcode-archive/blob/master/vendor/github.com/bitrise-io/xcode-project/xcodeproj/target.go#L50
and this after the dep update:
https://github.com/bitrise-steplib/steps-xcode-archive/blob/TOOL-20/vendor/github.com/bitrise-io/xcode-project/xcodeproj/target.go#L51
Basically, the if condition was extended with an "AND". To maintain the former behaviour, decided to set the bool to false, as with that the second part will be always true.

**Comment by richard-bogdan-bitrise** [comment]
@godrei: PR approved, pls merge.

File: utils/entitlements.go [hunk 2]
```diff
@@ -13,7 +13,7 @@ import (
 
 // ProjectEntitlementsByBundleID ...
 func ProjectEntitlementsByBundleID(pth, schemeName, configurationName string) (map[string]plistutil.PlistData, error) {
-	var scheme xcscheme.Scheme
+	var scheme *xcscheme.Scheme
```
Comment (godrei) [inline utils/entitlements.go]: pls check this comment: https://github.com/bitrise-io/xcode-project/pull/27#issue-250892062
the new version of the xcode-proj package makes this block much shorter

File: utils/entitlements.go [hunk 3]
```diff
@@ -7,46 +7,15 @@ import (
...
-		if err != nil {
-			if xcworkspace.IsSchemeNotFoundError(err) {
-				return nil, err
-			}
-			return nil, fmt.Errorf("failed to find scheme with name: %s in workspace: %s, error: %s", schemeName, pth, err)
-		}
-		schemeContainerDir = filepath.Dir(containerProject)
-	} else {
-		return nil, fmt.Errorf("unknown project extension: %s", filepath.Ext(pth))
+	scheme,schemeContainerDir,err := project.Scheme(pth,schemeName)
```
Comment (godrei) [inline utils/entitlements.go]: pls always use `gofmt` to format your code

Comment (richard-bogdan-bitrise) [reply utils/entitlements.go]: Updated code accordingly.

Comment (richard-bogdan-bitrise) [reply utils/entitlements.go]: Also added some minor fixes: there was package name collision and other minor fixes.

File: utils/entitlements.go (hunk 2 above)
Comment (richard-bogdan-bitrise) [reply utils/entitlements.go]: Updated code.

**Pull request merged**